libnmstate/netapplier.py

```diff
@@ -315,7 +318,35 @@ def assert_ifaces_state(ifaces_desired_state, ifaces_current_state):
 
         if iface_dstate != iface_cstate:
             raise DesiredStateIsNotCurrentError(
-                'desired: {}\ncurrent: {}'.format(iface_dstate, iface_cstate))
+                _format_desired_current_state_diff(iface_dstate,
+                                                   iface_cstate)
+            )
+
+
+def _format_desired_current_state_diff(desired_state, current_state):
+    pretty_desired_state = PrettyState(desired_state).yaml
+    pretty_current_state = PrettyState(current_state).yaml
+
+    diff = "".join(difflib.unified_diff(
+        pretty_desired_state.splitlines(True),
+        pretty_current_state.splitlines(True),
+        fromfile="desired",
+        tofile="current", n=0))
+    return (
+        '\n'
```

Comment:
How about using '\n'.join(['', 'desired', '=====', '{pretty_desired_state}', ...]).format(pretty_desired_state, ...)
Not a must, but it may look nicer without the multi '\n'.

Reply:
I like the multiple newline characters more... Another option would be to define a template at the beginning of the file

```
DIFF_TEMPLATE = """
desired
======
{desired]
...
```

Since we do not want to spend too much time on these features, maybe just keep it the way it is? :-)